packages/react/src/components/ConsumerControl.tsx

```diff
@@ -12,73 +11,93 @@ import {
 } from "./icons";
 
 import styles from "./ConsumerControl.module.css";
+import { useDataContext, useProviderContext } from "./context";
 
 interface ConsumerControlProps {
   consumer: Consumer;
   menuItems?: MenuItem[];
-  isLoading: boolean;
+  enableDeleteConsumer?: boolean;
 }
 
+export const ErrorContext = createContext<
+  [string | undefined, (error: string) => void]
+>([undefined, () => {}]);
+
+// 7 days
+const EXPIRY_PERIOD = 1000 * 60 * 60 * 24 * 7;
+
 const ConsumerControl = ({
   consumer,
   menuItems,
-  isLoading,
+  enableDeleteConsumer,
 }: ConsumerControlProps) => {
   const [edit, setEdit] = useState(false);
-  const [queryError, setQueryError] = useState<string>();
+  const [error, setError] = useState<string>();
   const [description, setDescription] = useState(consumer.description);
-  const { useConsumerDescriptionMutation, useRollKeyMutation } =
-    useQueryEngineContext();
-  const consumerDescriptionMutation = useConsumerDescriptionMutation();
-  const keyRollMutation = useRollKeyMutation();
-
-  useEffect(() => {
-    if (consumerDescriptionMutation.error) {
-      setQueryError(consumerDescriptionMutation.error.toString());
-      return;
-    }
-    setQueryError(undefined);
-  }, [
-    consumerDescriptionMutation.error,
-    consumerDescriptionMutation.isLoading,
-  ]);
-
-  useEffect(() => {
-    if (keyRollMutation.error) {
-      setQueryError(keyRollMutation.error.toString());
-      return;
-    }
-    setQueryError(undefined);
-  }, [keyRollMutation.error, keyRollMutation.isLoading]);
+  const [dataModel, setDataModel] = useDataContext();
+  const [descriptionUpdating, setDescriptionUpdating] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const provider = useProviderContext();
 
-  const handleMutationComplete = (error: unknown) => {
-    if (!error) {
-      setQueryError(undefined);
-      return;
-    }
-    if (error instanceof Error) {
-      setQueryError(error.message);
-      return;
+  const handleDescriptionSave = async () => {
+    try {
+      setDescriptionUpdating(true);
+      await provider.updateConsumerDescription(consumer.name, description);
+      const result = await provider.getConsumers();
+      setDataModel({
+        isFetching: dataModel?.isFetching,
+        consumers: result.data,
+      });
+
+      setEdit(false);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
```

Comment:
You don't need to actually add the type here
```suggestion
    } catch (err) {
```

Reply:
I can't use the .message property if I don't specify the type :(